tests/std/tests/GH_003572_fstream_seekp_0_cur/test.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <fstream>
+
+int main() {
+    using namespace std;
+
+    fstream f("test.txt", ios::in | ios::out | ios::trunc);
+
+    f << "123";
+
+    auto check = [&](string_view content) {
+        char buffer[4]{};
+        f.seekg(0);
+        f.read(buffer, 3);
+        assert(f && buffer == content);

```

Comment:
Suggestion: It's better to separately `assert(X);` and `assert(Y);` because if one of them fails, the line number will say which one it was. We try to follow this convention in our tests (not always perfectly).